kits/daisy/src/robot/leg.cpp

```diff
@@ -0,0 +1,172 @@
+#include "leg.hpp"
+#include <iostream>
+
+namespace hebi {
+
+using ActuatorType = hebi::robot_model::RobotModel::ActuatorType;
+using LinkType = hebi::robot_model::RobotModel::LinkType;
+
+Leg::Leg(double angle_rad, double distance, const Eigen::VectorXd& current_angles, const HexapodParameters& params, bool is_dummy, int index, LegConfiguration configuration)
+  : index_(index), stance_radius_(params.stance_radius_), body_height_(params.default_body_height_), spring_shift_(configuration == LegConfiguration::Right ? 3.75 : -3.75) // Nm
+{
+  kin_ = configuration == LegConfiguration::Left ?
+    hebi::robot_model::RobotModel::loadHRDF("left.hrdf") :
+    hebi::robot_model::RobotModel::loadHRDF("right.hrdf");
+  if (!kin_)
+  {
+    // Could not find HRDF files!
+    // TODO: handle this better so we don't segfault later...probably a factory
+    // for "leg"
+    std::cerr << "Could not find or load HRDF file for leg!" << std::endl;
+    assert("false");
+    return;
+  }
+
+  kin_->getMasses(masses_);
+
+  Matrix4d transform = Matrix4d::Identity();
+  Matrix3d rotate = AngleAxisd(angle_rad, Eigen::Vector3d::UnitZ()).matrix();
+  transform.topLeftCorner<3,3>() = rotate;
```

Comment:
:+1: for template overload of this (`<3, 3>`)

Reply:
I'm not always sloppy ;)